libraries/3rdparty/CMakeLists.txt

```diff
@@ -128,19 +128,27 @@ endif()
 
 # mbedtls - TLS protocol implementation
 if(EXISTS "${AFR_3RDPARTY_DIR}/mbedtls")
-    afr_glob_src(mbedtls_src DIRECTORY "${AFR_3RDPARTY_DIR}/mbedtls" RECURSE)
+    afr_glob_src(mbedtls_library DIRECTORY "${AFR_3RDPARTY_DIR}/mbedtls/library" RECURSE)
+    afr_glob_src(mbedtls_include DIRECTORY "${AFR_3RDPARTY_DIR}/mbedtls/include" RECURSE)
+
     # Build to static library to save some time.
     add_library(
         afr_3rdparty_mbedtls STATIC EXCLUDE_FROM_ALL
-        ${mbedtls_src}
+        ${mbedtls_library}
+        ${mbedtls_include}
+        "${AFR_3RDPARTY_DIR}/mbedtls_utils/mbedtls_utils.c"
         "${AFR_MODULES_ABSTRACTIONS_DIR}/pkcs11/mbedtls/threading_alt.h"
     )
+    # The "${AFR_3RDPARTY_DIR}/pkcs11" directory must be included before "${AFR_3RDPARTY_DIR}/mbedtls/include/mbedtls"
```

Comment:
Even after adding pkcs11 as a dependency, I was having issues getting it to have its include path come before the mbed tls directories. 

Other than that, I think it may be misleading to add pkcs11 as a dependency to the mbedtls module. MbedTLS doesn't have a dependency on pkcs11. It's more that when the pkcs11 module and the mbedtls module are used at the same time, the pkcs11 include path must come before the mbedtls module. This is due to the mbedtls directory having a "pkcs11.h" header file in it.

Considering your feedback, I'm not sure if my current code/comments gets this across clearly. One option to make this more clear could be to use a generator expression to only include the pkcs11 path when the pkcs11 target exists. It would look something like this:
```
    target_include_directories(
        afr_3rdparty_mbedtls PUBLIC 
        $<$<TARGET_EXISTS:3rdparty::pkcs11>:${AFR_3RDPARTY_DIR}/pkcs11>
    )
```
What are your thoughts on this?

Reply:
I remembered the discussion we had. The correct way to use mbedtls headers is `#include "mbedtls/xxx.h"` so that we don't get naming conflict, but however, some vendor code is doing this so it's preventing us to fix this. Let's leave this change as is here, and create a sim issue so we don't forget.